app/src/main/java/com/github/geohunt/app/ui/screens/maps/MapsViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package com.github.geohunt.app.ui.screens.maps
+
+import android.app.Application
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.viewmodel.initializer
+import androidx.lifecycle.viewmodel.viewModelFactory
+import com.github.geohunt.app.data.repository.AppContainer
+import com.github.geohunt.app.data.repository.ChallengeRepository
+import com.github.geohunt.app.model.Challenge
+import com.github.geohunt.app.utility.aggregateFlows
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class MapsViewModel(
+    private val challengeRepository: ChallengeRepository,
+) : ViewModel() {
+    private val _challenges: MutableStateFlow<List<Challenge>?> = MutableStateFlow(null)
+    val challenges: StateFlow<List<Challenge>?> = _challenges.asStateFlow()
+
+    fun updateFetchableChallenges(sectorHashes: List<String>) {
+        val aggregateChallengeFlow = sectorHashes.map { sectorHash ->
+            challengeRepository.getSectorChallenges(sectorHash)
+        }.aggregateFlows()
```

Comment:
You can place these lines inside the `viewModelScope.launch` block, so that the computation becomes cancelled if the coroutine is cancelled.

Reply:
Thank you for this comment. I have placed these lines of code inside the viewModelScope.launch